java/configuration/src/main/java/sleeper/configuration/properties/instance/S3InstanceProperties.java

```diff
@@ -40,33 +43,86 @@ public class S3InstanceProperties {
     private S3InstanceProperties() {
     }
 
+    /**
+     * Loads and validates instance properties from the config bucket of the given Sleeper instance.
+     *
+     * @param  s3Client   the S3 client
+     * @param  instanceId the Sleeper instance ID
+     * @return            the loaded instance properties
+     */
     public static InstanceProperties loadGivenInstanceId(AmazonS3 s3Client, String instanceId) {
         return InstanceProperties.createAndValidate(loadPropertiesGivenInstanceId(s3Client, instanceId));
     }
 
+    /**
+     * Loads instance properties from the config bucket of the given Sleeper instance, with no validation.
+     *
+     * @param  s3Client   the S3 client
+     * @param  instanceId the Sleeper instance ID
+     * @return            the loaded instance properties
+     */
     public static InstanceProperties loadGivenInstanceIdNoValidation(AmazonS3 s3Client, String instanceId) {
         return InstanceProperties.createWithoutValidation(loadPropertiesGivenInstanceId(s3Client, instanceId));
     }
 
+    /**
+     * Loads instance properties from the given S3 bucket.
```

Comment:
Do we need to mention this also includes validation?

Reply:
Done, thanks